python/pyspark/sql/functions/builtin.py

```diff
@@ -10926,6 +10928,9 @@ def substring(str: "ColumnOrName", pos: int, len: int) -> Column:
     .. versionchanged:: 3.4.0
         Supports Spark Connect.
 
+    .. versionchanged:: 4.0.0
```

Comment:
let's move it under parameters section, you may refer to

https://github.com/apache/spark/blob/bb1f026da0e3193e0cfad2001c7a8cab59a5290a/python/pyspark/sql/functions/builtin.py#L6216-L6220